src/components/JUPTable/JUPTable.tsx

```diff
@@ -57,92 +49,136 @@ function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
 
 type Order = "asc" | "desc";
 
-function getComparator<Key extends keyof ITransaction>(
-  order: Order,
-  orderBy: Key
-): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+function getComparator(order: Order, orderBy: "toString" | "valueOf"): (a: ITableRow, b: ITableRow) => number {
   return order === "desc" ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
 }
 
 interface IEnhancedTableProps {
-  // onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Data) => void;
   onRequestSort: (event: React.MouseEvent<unknown>, property: string) => void;
   order: Order;
   orderBy: string;
-  headCells: Array<IHeadCellProps>;
+  headCells?: Array<IHeadCellProps>;
 }
 
 const EnhancedTableHead: React.FC<IEnhancedTableProps> = ({ onRequestSort, order, orderBy, headCells }) => {
-  const createSortHandler = (property: keyof Data) => (event: React.MouseEvent<unknown>) => {
-    onRequestSort(event, property);
-  };
+  const createSortHandler = useCallback(
+    (event: React.MouseEvent<unknown>, property: string) => {
+      onRequestSort(event, property);
+    },
+    [onRequestSort]
+  );
+
+  const HeadCellsMemo = useMemo(() => {
```

Comment:
i'm considering how to add a 'no default sort order' to this as there might be times where none is wanted. FYI